src/nifc/genexprs.nim

```diff
@@ -204,6 +204,9 @@ proc genAddr(c: var GeneratedCode; n: var Cursor) =
   genx c, n
   if arrType.typeKind == ArrayT and not c.m.isImportC(arrType):
     c.add ".a"
+    c.add BracketLe
+    c.add "0"
+    c.add BracketRi
```

Comment:
What you need to do here is to detect if you're within a function call and that function is imported from C and then deduce from this fact that the address operator needs to be special cased.

In other words, inside `genCall` add special logic for `AddrC`.